src/encoder.rs

```diff
@@ -15,9 +15,11 @@ use crate::systematic_constants::num_lt_symbols;
 use crate::systematic_constants::num_pi_symbols;
 use crate::systematic_constants::{calculate_p1, systematic_index};
 use crate::ObjectTransmissionInformation;
+use serde::{Deserialize, Serialize};
 
 pub const SPARSE_MATRIX_THRESHOLD: u32 = 250;
 
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, PartialOrd, Eq, Ord, Hash)]
```

Comment:
Please remove Hash, Ord, and PartialOrd. The rest of the traits seem fine, but similarly to how those can't be added to Decoder, I think Encoder may include fields which don't implement `Ord` in the future.